leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisRegistrationStore.java

```diff
@@ -768,4 +754,273 @@ public void run() {
+         * Default value is {@literal LOCK#EP#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal LOCK:EP:}.
+         */
+        public Builder setEndpointLockPrefix(String endpointLockPrefix) {
+            this.endpointLockPrefix = endpointLockPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key prefix for observation token lookup.
+         * <p>
+         * Default value is {@literal OBS#TKN#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal OBS:TKN:}.
+         */
+        public Builder setObservationTokenPrefix(String observationTokenPrefix) {
+            this.observationTokenPrefix = observationTokenPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key prefix for observation tokens list lookup by registration ID.
+         * <p>
+         * Default value is {@literal TKNS#REGID#}. Should not be {@code null} or empty. Leshan v1.x used
+         * {@literal TKNS:REGID:}.
+         */
+        public Builder setObservationTokensByRegistrationIdPrefix(String observationTokensByRegistrationIdPrefix) {
+            this.observationTokensByRegistrationIdPrefix = observationTokensByRegistrationIdPrefix;
+            return this;
+        }
+
+        /**
+         * Set the key for expiration key lookup. It is a sorted set used for registration expiration (expiration date,
+         * endpoint).
+         * <p>
+         * Default value is {@literal EXP#EP}. Should not be {@code null} or empty. Leshan v1.x used {@literal EXP:EP}.
+         */
+        public Builder setEndpointExpirationKey(String endpointExpirationKey) {
+            this.endpointExpirationKey = endpointExpirationKey;
+            return this;
+        }
+
+        /**
+         * Set the clean period in seconds.
+         * <p>
+         * Default value is {@literal 60 seconds}.
+         */
+        public Builder setCleanPeriod(long cleanPeriod) {
+            this.cleanPeriod = cleanPeriod;
+            return this;
+        }
+
+        /**
+         * Set the maximum number to clean in a clean period.
+         * <p>
+         * Default value is {@literal 500}.
+         */
+        public Builder setCleanLimit(int cleanLimit) {
+            this.cleanLimit = cleanLimit;
+            return this;
+        }
+
+        /**
+         * Set the grace period in seconds.
+         * <p>
+         * Default value is {@literal 0 seconds}.
+         */
+        public Builder setGracePeriod(long gracePeriod) {
+            this.gracePeriod = gracePeriod;
+            return this;
+        }
+
+        public Builder setSchedExecutor(ScheduledExecutorService schedExecutor) {
+            this.schedExecutor = schedExecutor;
+            return this;
+        }
+
+        public Builder setLock(JedisLock lock) {
+            this.lock = lock;
+            return this;
+        }
+
+        public Builder setRegistrationSerDes(RegistrationSerDes registrationSerDes) {
```

Comment:
Maybe something like :

Set `{@link RegistrationSerDes}` instance used to serialize/de-serialize registration to/from this store.